python/lsst/ctrl/bps/htcondor/htcondor_service.py

```diff
@@ -1952,19 +1962,31 @@ def _create_periodic_release_expr(memory, multiplier, limit):
     ----------
     memory : `int`
         Requested memory in MB.
-    multiplier : `float`
+    multiplier : `float` or None
         Memory growth rate between retires.
     limit : `int`
         Memory limit.
+    additional_expr : `str`, optional
+        Expression to add to periodic_release.  Defaults to empty string.
 
     Returns
     -------
     expr : `str`
         A string representing an HTCondor ClassAd expression for releasing jobs
         which have been held due to exceeding the memory requirements.
     """
-    is_retry_allowed = "NumJobStarts <= JobMaxRetries"
-    was_below_limit = f"min({{int({memory} * pow({multiplier}, NumJobStarts - 1)), {limit}}}) < {limit}"
+    _LOG.debug(
+        "periodic_release: memory: %s, multiplier: %s, limit: %s, additional_expr: %s",
+        memory,
+        multiplier,
+        limit,
+        additional_expr,
+    )
+
+    # ctrl_bps sets multiplier to None in the GenericWorkflow if
+    # memoryMultiplier <= 1, but checking value just in case.
+    if (not multiplier or multiplier <= 1) and not additional_expr:
+        return ""
```

Comment:
Checking if `multiplier` is smaller than 1 here seems to be unnecessary/redundant. You essentially repeat this test in line 2006). If I got the intentions right, you'd like to exit early when neither `multiplier` nor `additional_expr` is set. So something like
```python
if not multiplier and not additional_expr:
    return ""
```
would suffice.

Reply:
If remove that check and multiplier is 1 and additional_expr is None, then would get an expr with JobStatus == 5 && NumJobStarts <= JobMaxRetries (automatically release every held job if still have retries left).  Leaving the check as is, the returned expr is the empty string.  I don't think we want to add that kind of automatic release unless specified by user in the additional_expr.

Reply:
Right, we don't want to release every held job so some additional changes would be needed. Let's leave it as is then.